DashKit/DashPumpManager.swift

```diff
@@ -308,14 +294,21 @@ public class DashPumpManager: PumpManager {
     }
     
     public func discardPod(completion: @escaping (PodCommResult<Bool>) -> ()) {
+        
         podCommManager.discardPod { (result) in
             self.mutateState({ (state) in
+                let now = self.dateGenerator()
+                state.unfinalizedBolus?.cancel(at: now)
+                state.unfinalizedTempBasal?.cancel(at: now)
+                state.suspendState = .suspended(now)
                 state.podActivatedAt = nil
                 state.lastStatusDate = nil
                 state.reservoirLevel = nil
                 state.podTotalDelivery = nil
             })
-            completion(result)
+            self.finalizeAndStoreDoses(completion: { (_) in
```

Comment:
Do we care about the `Error?` from the `finalizeAndStoreDoses` completion?

Reply:
`finalizeAndStoreDoses` logs any errors, and avoids clearing out pendingDoses on error, so these doses would stick around to be stored later if necessary, when a new pod was attached.